core/modules/catalog/components/ProductVideo.ts

```diff
@@ -0,0 +1,48 @@
+export const ProductVideo = {
+  name: 'ProductVideo',
+  props: {
+    url: {
+      type: String,
+      required: true
+    },
+    id: {
+      type: String,
+      required: true
+    },
+    type: {
+      type: String,
+      required: true
+    },
+    index: {
```

Comment:
What is the purpose of this  property? 
If it is emitted later in the code would be good to have a default value instead of undefined

Reply:
It started out as optional because I wasn't sure if the ProductVideo component would always have an index. But you're right, I'm using it later without checking if its set. I gave it a default value of `0` now.